02-express-tutorial/controlers/people.js

```diff
@@ -0,0 +1,59 @@
+let { people } = require("../data");
+
+const getPeople = (req, res) => {   
+    res.status(200).json({ success: true, data: people });
+}
+const createPerson = (req, res) => {
+    const { name } = req.body;
+    if(!name){
+        return res
+            .status(400)
+            .json({ success: false, msg: "Please provide name value" });
+    }
+    res.status(201).json({ success: true, person: name }); 
+}
+const createPersonPostman =  (req, res) => {
+    const { name } = req.body;
+    if(!name){
+        return res
+            .status(400)
+            .json({ success: false, msg: "Please provide name value" });
+    }
+    res.status(201).json({ success: true, data: [...people, name] }); 
+}
+const updatePerson = (req, res) => {
+    const { id } = req.params;
+    const { name } = req.body;
+    const person = people.find((person) => person.id === Number(id));
+    if(!person){
+        return res
+            .status(404)
+            .json({ success: false, msg: `no person with id ${id}` });
+    }
+    const newPeople = people.map((person) => {
+    if(person.id === Number(id)){
+        person.name = name;
+    }
+    return person;
+    });
+    res.status(200).json({ success: true, data: newPeople });
+}
+const deletePerson = (req, res) => {
+    const person = people.find((person) => person.id === Number(req.params.id));
+        if(!person){
+            return res
+            .status(404)
+            .json({ success: false, msg: `no person with id ${req.params.id}` });
+        }
+    const newPeople = people.filter(
+        (person) => person.id !== Number(req.params.id)
+    );
+    return res.status(200).json({ success: true, data: newPeople });
```

Comment:
Nitpick (doesn't require change):
This can be simplified using the `findIndex` and `splice` Array methods:
```js
const personId = parseInt(req.params.id, 10);

const personIndex = people.findIndex(p => p.id === personId);

if (personIndex < 0) {
  return res.status(404).json({ message: "Person not found" });
}

people.splice(personIndex, 1);
 
return res.status(200).json({ success: true, data: newPeople });
```